homeassistant/components/onewire/entity.py

```diff
@@ -35,19 +35,26 @@ def __init__(
         description: OneWireEntityDescription,
         device_id: str,
         device_info: DeviceInfo,
-        device_file: str,
+        device_file: str | None,
         owproxy: protocol._Proxy,
+        family: str | None = None,
     ) -> None:
         """Initialize the entity."""
         self.entity_description = description
         self._last_update_success = True
         self._attr_unique_id = f"/{device_id}/{description.key}"
         self._attr_device_info = device_info
         self._device_file = device_file
-        self._state: int | float | None = None
-        self._value_raw: float | None = None
+        self._state: int | float | str | None = None
+        self._value_raw: float | str | None = None
         self._owproxy = owproxy
 
+        if family == "01":
+            self._attr_name = "Serial Number"
```

Comment:
What is the purpose of this sensor?

Reply:
In my situation, I use them for two purposes:
1) identifying which 1wire-to-USB converter is connected and which is not (I have a dozen of them)
2) a hardware key to identify "logged-in" users

Reply:
Maybe reduce the current PR to simply create the device (ie. add "01" to DEVICE_SUPPORT) without entity creation.
And you can look at adding the binary sensor in a follow-up PR